app/Jobs/RequeuePendingQsBatchesJob.php

```diff
@@ -18,23 +18,23 @@ public function __construct() {
 
     public function handle(): void
     {
-        DB::transaction(function () {
-            tap(QsBatch::where([
+        $failedBatches = [];
+        DB::transaction(function () use (&$failedBatches) {
+            $failedBatches = QsBatch::where([
                 ['processing_attempts', '>=', $this->markFailedAfter],
                 ['failed', '=', false],
-            ]))
-                ->update(['failed' => true])
-                ->get()
-                ->each(function ($batch, $index) {
-                    report("QsBatch with ID ".$batch->id."was marked as failed.");
-                });
-
-            $threshold = Carbon::now()->subtract(new \DateInterval($this->pendingTimeout));
-            QsBatch::where([['pending_since', '<>', null], ['pending_since', '<', $threshold]])
-                ->increment(
-                    'processing_attempts', 1,
-                    ['pending_since' => null, 'done' => 0]
-                );
+            ])->get()->pluck('id');
+            QsBatch::whereIn('id', $failedBatches)->update(['failed' => true]);
         });
+        foreach ($failedBatches as $batchId) {
```

Comment:
I moved this out of the transaction so we know everything has been comitted before we send out the errors.